### PR TITLE
added 2 items to ItemCooldowns

### PR DESCRIPTION
**Pull request by VeritasDL**
added Rouge Sword and Ink Wand

### CODE REVIEW

**Review by kevinthegreat1 — APPROVED** [review]

**Review by kevinthegreat1 — COMMENTED** [review]
You have a lot of duplicate code, can you use a method please.

**Comment by VeritasDL** [comment]
pardon my tism, this might be slightly better?

File: src/main/java/de/hysky/skyblocker/skyblock/item/ItemCooldowns.java [hunk 1]
```diff
@@ -95,55 +107,40 @@ public static void afterBlockBreak(World world, PlayerEntity player, BlockPos po
...
-				ITEM_COOLDOWNS.put(INK_WAND_ID, new CooldownEntry(30000));
-			}
+		if (!SkyblockerConfigManager.get().uiAndVisuals.itemCooldown.enableItemCooldowns)
+			return TypedActionResult.pass(ItemStack.EMPTY);
+		String usedItemId = ItemUtils.getItemId(player.getMainHandStack());
+		switch (usedItemId) {
+			case GRAPPLING_HOOK_ID:
+				if (player.fishHook != null) {
+					handleItemCooldown(GRAPPLING_HOOK_ID, 2000, !isWearingBatArmor(player));
+				}
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/item/ItemCooldowns.java:117]:
```suggestion
				handleItemCooldown(GRAPPLING_HOOK_ID, 2000, player.fishHook != null && !isWearingBatArmor(player));
```

File: src/main/java/de/hysky/skyblocker/skyblock/item/ItemCooldowns.java [hunk 2]
```diff
@@ -80,6 +80,18 @@ private static int getCooldown4Foraging() {
...
+		if (!isOnCooldown(itemId) && additionalCondition) {
+			ITEM_COOLDOWNS.put(itemId, new CooldownEntry(cooldownTime));
+		}
+	}
+
+	// Overloaded method for cases without additional conditions
+	private static void handleItemCooldown(String itemId, int cooldownTime) {
+		handleItemCooldown(itemId, cooldownTime, true);
+	}
+
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/item/ItemCooldowns.java:94]: It doesn't make sense to place these two methods here. Maybe after `onItemInteract`?

File: src/main/java/de/hysky/skyblocker/skyblock/item/ItemCooldowns.java [hunk 3]
```diff
@@ -95,55 +107,40 @@ public static void afterBlockBreak(World world, PlayerEntity player, BlockPos po
...
-			}
-		}
-		if (usedItemId.equals(INK_WAND_ID)) {
-			if (!isOnCooldown(INK_WAND_ID)) {
-				ITEM_COOLDOWNS.put(INK_WAND_ID, new CooldownEntry(30000));
-			}
+		if (!SkyblockerConfigManager.get().uiAndVisuals.itemCooldown.enableItemCooldowns)
+			return TypedActionResult.pass(ItemStack.EMPTY);
+		String usedItemId = ItemUtils.getItemId(player.getMainHandStack());
+		switch (usedItemId) {
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/item/ItemCooldowns.java:113]: Can you sort this switch by the time while you're at it? You also have two entries for `30000`.

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]

**Comment by VeritasDL** [comment]
weeeee

**Review by kevinthegreat1 — APPROVED** [review]
Looks great, thanks.

**Pull request merged**